src/clj/rems/db/user_settings.clj

```diff
@@ -1,21 +1,22 @@
 (ns rems.db.user-settings
   (:require [clojure.string :as str]
-            [medley.core :refer [map-keys]]
             [rems.common.util :refer [+email-regex+]]
             [rems.config :refer [env]]
             [rems.db.core :as db]
             [rems.json :as json]
             [schema.coerce :as coerce]
-            [schema.core :as s]))
+            [schema.core :as s])
+  (:import (org.joda.time DateTime)))
 
 (defn- default-settings []
   {:language (:default-language env)
    :notification-email nil})
 
 ;; TODO should this be in schema-base?
 (s/defschema UserSettings
-  {:language s/Keyword
-   :notification-email (s/maybe s/Str)})
+  {(s/optional-key :language) s/Keyword
+   (s/optional-key :notification-email) (s/maybe s/Str)
+   (s/optional-key :ega) {:api-key-expiration-date DateTime}})
```

Comment:
This refactor could be split out to its own PR and discussed separately. I'm slightly concerned that we won't guarantee the presence of any of these keys, even though the UI expects `:language` and `:notification-email`.

Reply:
The API did not guarantee them before either?

Reply:
No, the keys are required in the API response, see
- how rems.api.user-settings usess the user-settings/UserSettings schema
- https://rems-dev.rahtiapp.fi/swagger-ui/index.html#/user-settings/get_api_user_settings

Reply:
How is it possible? It used to reference `PartialUserSettings` schema where they were optional.

Reply:
Ah, I misread. Only PUT referenced them.

Reply:
Since these values come from the `default-settings` if they don't exist we could force the presence of the keys like before.

Reply:
In here the original problem is doing it differently, and skipping separate read and write models. Cleaning up in a backwards compatible way is difficult. But I could introduce separate read & write models.

Reply:
Yeah that's why I wanted to split this out into a separate PR, we need to do this carefully. I like the schema simplification. Maybe we can have something like `(defn UserSettingsInApi (assoc UserSettings :language s/Keyword :notification-email s/Str))` instead of the `PartialUserSettings` hack.

Reply:
We can have separate DB and API schemas as the others?

Reply:
Yeah can just do full decoupling

Reply:
I'll do that and split the small parts out.

Reply:
This PR is now updated with what I think is the separate db, read and write models with backwards compatibility. I have to look at it with fresh eyes still. I'll harvest them to another PR next.

Reply:
https://github.com/CSCfi/rems/pull/2606